test/Microsoft.AspNet.Razor.Test/Parser/CSharp/CSharpBlockTest.cs

```diff
@@ -736,6 +737,231 @@ public void ParsersCanNestRecursively()
+                                        factory.Markup(" foo='").With(SpanCodeGenerator.Null),
+                                        factory.Markup("abc").With(new LiteralAttributeCodeGenerator(new LocationTagged<string>(string.Empty, 12, 0, 12), new LocationTagged<string>("abc", 12, 0, 12))),
+                                        factory.Markup(" "),
+                                        factory.Markup("@").With(SpanCodeGenerator.Null),
+                                        factory.Markup("@"),
+                                        factory.Markup(" def").With(new LiteralAttributeCodeGenerator(new LocationTagged<string>(" ", 18, 0, 18), new LocationTagged<string>("def", 19, 0, 19))),
+                                        factory.Markup("'").With(SpanCodeGenerator.Null)),
+                                factory.Markup(" />").Accepts(AcceptedCharacters.None))),
+                            factory.EmptyCSharp().AsStatement(),
+                            factory.MetaCode("}").Accepts(AcceptedCharacters.None))
+                    },
+                    {
+                        "{<span foo='@@@DateTime.Now' />}",
+                        new StatementBlock(
+                            factory.MetaCode("{").Accepts(AcceptedCharacters.None),
+                            new MarkupBlock(
+                                new MarkupTagBlock(
+                                    factory.Markup("<span"),
+                                    new MarkupBlock(
+                                        new AttributeBlockCodeGenerator("foo", new LocationTagged<string>(" foo='", 6, 0, 6), new LocationTagged<string>("'", 27, 0, 27)),
+                                        factory.Markup(" foo='").With(SpanCodeGenerator.Null),
+                                        factory.Markup("@").With(SpanCodeGenerator.Null),
+                                        factory.Markup("@"),
+                                        new MarkupBlock(
+                                            new DynamicAttributeBlockCodeGenerator(new LocationTagged<string>(string.Empty, 14, 0, 14), 14, 0, 14),
+                                            new ExpressionBlock(
+                                                factory.CodeTransition(),
+                                                factory.Code("DateTime.Now")
+                                                    .AsImplicitExpression(CSharpCodeParser.DefaultKeywords)
+                                                    .Accepts(AcceptedCharacters.NonWhiteSpace))),
+                                        factory.Markup("'").With(SpanCodeGenerator.Null)),
+                                factory.Markup(" />").Accepts(AcceptedCharacters.None))),
+                            factory.EmptyCSharp().AsStatement(),
+                            factory.MetaCode("}").Accepts(AcceptedCharacters.None))
+                    },
+                    {
+                        "{<span foo='@DateTime.Now @@' />}",
```

Comment:
Add another variation of this test case but without the space between `.Now` and `@@` (do same for document tests)